installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -616,6 +627,7 @@ public static ActionResult SetNetworkErrorProperties(Session session)
                 RollbarReport.Error(msg, session);
             }
 
+            session.DoAction("GAReportUserNetwork");
```

Comment:
Interesting!  So, you can run "immediate" actions from a "deferred" action, and then use the `session` variables.  Maybe we can use to reduce the list of variables that we set in the `CustomActionData`?

My concern with this is, that with this set-up both the `stage/finished/network` and the `stage/finished/failure` messages will be sent. @Naatan Is that ok?

Reply:
This method is actually run as immediate. The flow is a little different, it's declared [here](https://github.com/ActiveState/cli/blob/master/installers/msi-language/Product.p.wxs#L224) and eventually called via the Custom fatal error dialog [here](https://github.com/ActiveState/cli/blob/master/installers/msi-language/CustomFatalError.wxs#L37). I had to do it this way as you can only schedule one action to occur with `OnExit="error"`

Reply:
Ah! Maybe I get it know.  So, `OnExit=error` we are just invoking the `SetNetworkErrorProperties` custom action which then is calling the `CustomFatalError` dialog.

However, the `CustomFatalError` dialog is still calling `GAReportFailure` when the user clicks the `"Finish"` button https://github.com/ActiveState/cli/blob/302fe08257775318e0602bd5348bb1b8024df085/installers/msi-language/Product.p.wxs#L130.

But don't we then invoke `GAReportUserNetwork` every time, even if `NETWORK_ERROR` is empty?

Reply:
Yes you are right about the `GaReportUserNetwork`. That should be now fixed.

I've decoupled the `GAReportFailure` from the Finish button on the exit dialog and instead it is now triggered from `SetNetworkProperties` if there is no network error.